tests/integration/util/lib.go

```diff
@@ -0,0 +1,418 @@
+package util
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+)
+
+/* Utility providing support primarily for k8ssandra integration tests */
+
+// AnnotateResult model for release annotations applied
+type AnnotateResult struct {
+	IsReleaseNamespaceApplied bool
+	IsReleaseNameApplied      bool
+}
+
+// CRDIdentity test model for Custom Resource Identifier
+type CRDIdentity struct {
+	CRDName         string
+	CRDHookName     string
+	CRDResourceName string
+	CRDMetaName     string
+}
+
+// ClusterIdentity test model for cluster
+type ClusterIdentity struct {
+	ClusterName            string
+	RenderedCassdcTemplate string
+}
+
+// CreateOperatorIdentity factory function for Operator Identity
+func CreateOperatorIdentity() CRDIdentity {
+
+	return CRDIdentity{
+		CRDName:         "crd/cassandradatacenters.cassandra.datastax.com",
+		CRDHookName:     "cass-operator-webhook",
+		CRDResourceName: "cassandradatacenters.cassandra.datastax.com"}
+}
+
+// CreateClusterIdentity performs construction of cluster identity
+// includes checks for required chart used along with associated rendered template.
+func CreateClusterIdentity(t *testing.T, kubeOptions *k8s.KubectlOptions,
+	clusterName string, releaseName string) ClusterIdentity {
+
+	var renderedMap map[string]interface{}
+	helmChartPath, err := filepath.Abs("../../charts/k8ssandra-cluster")
+	require.NoError(t, err)
+
+	options := &helm.Options{
+		SetStrValues:   map[string]string{"name": kubeOptions.Namespace, "clusterName": clusterName},
+		KubectlOptions: kubeOptions,
+	}
+
+	renderedOutput := helm.RenderTemplate(
+		t, options, helmChartPath, releaseName,
+		[]string{"templates/cassdc.yaml"},
+	)
+	jsonRendered, err := yaml.YAMLToJSON([]byte(renderedOutput))
+	require.NoError(t, json.Unmarshal(jsonRendered, &renderedMap))
+
+	return ClusterIdentity{
+		ClusterName:            clusterName,
+		RenderedCassdcTemplate: renderedOutput,
+	}
+}
+
+// Annotate of release name, and release namespace, scoped to namespace
+func Annotate(t *testing.T, helmOptions *helm.Options, releaseName string) *AnnotateResult {
```

Comment:
I am not sure I understand why we need this function. Or to ask another way, why are we overwriting annotations that Helm adds?

Reply:
Updating with a summary comment to address why this was created.